### PR TITLE
[Qwen3.5] Update model Readme and integrate XLML test scripts

### PR DESCRIPTION
**Pull request by Rohan-Bierneni**
# Description

Add qwen3.5 model to qwen moe readme. Also add scripts to run qwen3.5 in daily xlml tests.

If the change fixes a bug or a Github issue, please include a link, e.g.,:
FIXES: b/123456

# Tests

Will run test on xpk v5p-128 cluster.

xpk cmd: https://paste.googleplex.com/4783924076216320

Output logs: https://paste.googleplex.com/6553904014819328

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [X] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [X] I have necessary comments in my code, particularly in hard-to-understand areas.
- [X] I have run end-to-end tests tests and provided workload links above if applicable.
- [X] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

**Review by entrpn — APPROVED** [review]

**Review by hengtaoguo — APPROVED** [review]

File: tests/end_to_end/tpu/qwen/moe/qwen3.5-35b-a3b/1_test_qwen3.5_35b_a3b.sh [hunk 1]
```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+
+# This file is documentation for how to get started with Qwen3 Next.
```
Comment (shuningjin) [inline tests/end_to_end/tpu/qwen/moe/qwen3.5-35b-a3b/1_test_qwen3.5_35b_a3b.sh]: nit: Qwen3 Next -> Qwen3.5-35B 

(same for `1_test_qwen3.5-397b-a17b.sh`)

File: tests/end_to_end/tpu/qwen/moe/qwen3.5-35b-a3b/2_test_qwen3.5_35b_a3b.sh [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+# default golden_logits_path=/deps/tests/assets/golden_logits/golden_data_{MODEL_NAME}.jsonl, copied from gs://maxtext-test-assets/golden_data_${MODEL_NAME}.jsonl
+GOLDEN_LOGITS_DISK_LOCATION="/deps/tests/assets/golden_logits/golden_data_${MODEL_NAME}.jsonl"
+if [ ! -f "${GOLDEN_LOGITS_DISK_LOCATION}" ]; then
+  GOLDEN_LOGITS_PATH="gs://maxtext-test-assets/golden_data_${MODEL_NAME}.jsonl"
+  GOLDEN_LOGITS_DISK_LOCATION=/tmp/golden_data.jsonl
+  gcloud storage cp ${GOLDEN_LOGITS_PATH} ${GOLDEN_LOGITS_DISK_LOCATION}
+fi
+
+python3 -m tests.utils.forward_pass_logit_checker ${MAXTEXT_PKG_DIR}/configs/base.yml base_output_directory=${BASE_OUTPUT_PATH} run_name=forward_logits_check load_parameters_path=${SCANNED_CKPT_PATH} scan_layers=true attention=dot_product per_device_batch_size=1 model_name=${MODEL_NAME} max_prefill_predict_length=4 max_target_length=4 async_checkpointing=false sparse_matmul=True ici_fsdp_parallelism=1 ici_expert_parallelism=-1 weight_dtype=float32 dtype=float32 activations_in_float32=true matmul_precision=highest float32_logits=true float32_qk_product=true --golden_logits_path=${GOLDEN_LOGITS_DISK_LOCATION} --atol=1.5 --rtol=1.5 --max_kl_div=0.2
```
Comment (shuningjin) [inline tests/end_to_end/tpu/qwen/moe/qwen3.5-35b-a3b/2_test_qwen3.5_35b_a3b.sh]: In the past, we used `${MAXTEXT_PKG_DIR}/configs/base.yml`. The current scripts are updated to `${MAXTEXT_CONFIGS_DIR:-${MAXTEXT_REPO_ROOT:-$PWD}/configs/base.yml`, 

https://github.com/AI-Hypercomputer/maxtext/blob/194fb8aaaa85c6732ae1d5c28f250f243d235b27/tests/end_to_end/tpu/deepseek/v3.2-671b/2_test_deepseek.sh#L52

Please update all occurrences accordingly.

File: tests/end_to_end/tpu/qwen/moe/qwen3.5-35b-a3b/1_test_qwen3.5_35b_a3b.sh [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+#    Scanned format is better for training; unscanned format is better for decoding.
+# 2. Run logit check, pre-training, fine-tuning, and decoding.
+# ---
+# Example Usage:
+#
+# # (Required) Path to the converted MaxText checkpoint
+# export MAXTEXT_CHECKPOINT_PATH=gs://path/to/converted_ckpt/0/items/
+#
+# # (Optional) Override the default HF model
+# export HF_MODEL_PATH=MyCustom/Qwen3-variant
```
Comment (shuningjin) [inline tests/end_to_end/tpu/qwen/moe/qwen3.5-35b-a3b/1_test_qwen3.5_35b_a3b.sh]: These variables seems not used in the conversion command?

(same for `1_test_qwen3.5-397b-a17b.sh`)

File: tests/end_to_end/tpu/qwen/moe/qwen3.5-397b-a17b/1_test_qwen3.5-397b-a17b.sh [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+# ---
+# Example Usage:
+#
+# # (Required) Path to the converted MaxText checkpoint
+# export MAXTEXT_CHECKPOINT_PATH=gs://path/to/converted_ckpt/0/items/
+#
+# # (Optional) Override the default HF model
+# export HF_MODEL_PATH=MyCustom/Qwen3-variant
+#
+# bash tests/end_to_end/tpu/qwen/moe/qwen3.5-35b-a3b/1_test_qwen3.5_35b_a3b.sh
```
Comment (shuningjin) [inline tests/end_to_end/tpu/qwen/moe/qwen3.5-397b-a17b/1_test_qwen3.5-397b-a17b.sh]: nit: `bash tests/end_to_end/tpu/qwen/moe/qwen3.5-397b-a17b/1_test_qwen3.5-397b-a17b.sh‎`

File: tests/end_to_end/tpu/qwen/moe/run_qwen_moe.md [hunk 5]
```diff
@@ -21,19 +27,19 @@ To get started, you first need a MaxText-compatible checkpoint.
...
 
-2.  **Convert the Checkpoint**: Run the `convert_qwen3_moe.py` script to convert the downloaded Hugging Face weights into the Orbax format required by MaxText.
+2.  **Convert the Checkpoint**: Run the `convert_qwen3.5_35b_scanned.py` script to convert the downloaded Hugging Face weights into the Orbax format required by MaxText.
 
     ```
-    python3 -m maxtext.checkpoint_conversion.standalone_scripts.convert_qwen3_moe\
-      --base_model_path /path/to/qwen3_hf_checkpoint\
-      --maxtext_model_path gs://your-gcs-bucket/qwen3_maxtext_ckpt\
-      --model_size <qwen3-30b-a3b|qwen3-235b-a22b|qwen3-480b-a35b>
-
```
Comment (shuningjin) [inline tests/end_to_end/tpu/qwen/moe/run_qwen_moe.md:36]: Can you retain the example for qwen3, and add qwen3.5 on top, perhaps as a sub-section?

File: tests/end_to_end/tpu/qwen/moe/run_qwen_moe.md [hunk 6]
```diff
@@ -133,4 +139,34 @@ export MAXTEXT_CHECKPOINT_PATH=gs://your-gcs-bucket/qwen3-480b-a35b_maxtext_ckpt
...
+### Qwen3.5-35B-A3B
+
+Bash
+
+```
+# Set the required path to your converted MaxText checkpoint
+export MAXTEXT_CHECKPOINT_PATH=gs://your-gcs-bucket/qwen3.5-35b-a3b_maxtext_ckpt/0/items/
+
+# (Optional) Set the path to your local Hugging Face checkpoint
+# export HF_MODEL_PATH=/path/to/local/qwen3.5-35b-a3b_hf_checkpoint
```
Comment (shuningjin) [inline tests/end_to_end/tpu/qwen/moe/run_qwen_moe.md]: are these variables used in the script?

**Review by shuningjin — COMMENTED** [review]
Thank you for adding test and doc! Some minor comments.

File: tests/end_to_end/tpu/qwen/moe/qwen3.5-35b-a3b/2_test_qwen3.5_35b_a3b.sh (hunk 2 above)
Comment (Rohan-Bierneni) [reply tests/end_to_end/tpu/qwen/moe/qwen3.5-35b-a3b/2_test_qwen3.5_35b_a3b.sh]: Resolved.

File: tests/end_to_end/tpu/qwen/moe/run_qwen_moe.md (hunk 5 above)
Comment (Rohan-Bierneni) [reply tests/end_to_end/tpu/qwen/moe/run_qwen_moe.md:36]: I have made this change but also made the addition of showcasing legacy checkpoint conversion for qwen3 & checkpoint util for qwen3.5. Also included link to checkpoint util doc. This is how the change looks:

* * * * *

Checkpoint Conversion
---------------------

To get started, you first need a MaxText-compatible checkpoint.

1.  **Download the Model**: Download the official model from Hugging Face. You can use a tool like `hf_transfer` for a fast download.

    ```
    # Example for Qwen3.5-35B-A3B
    hf_transfer download Qwen/Qwen3.5-35B-A3B --local-dir /path/to/qwen3.5_35b_hf_checkpoint
    ```

    ```
    # Example for Qwen3-235B-A22B
    hf_transfer download Qwen/Qwen3-235B-A22B-Thinking-2507 --local-dir /path/to/qwen3_hf_checkpoint
    ```

2.  **Convert the Checkpoint**: 

  - **Checkpointing Util**: Supports bidirection conversion from MaxText <-> HF. For more info look at [convert_checkpoint](https://github.com/AI-Hypercomputer/maxtext/blob/main/docs/guides/checkpointing_solutions/convert_checkpoint.md) doc.

    ```
    JAX_PLATFORMS=cpu python3 -m maxtext.checkpoint_conversion.to_maxtext src/maxtext/configs/base.yml \
        model_name=qwen3.5-35b-a3b \
        base_output_directory=gs://your-gcs-bucket/qwen3.5_35b_maxtext_ckpt \
        hf_access_token=${HF_TOKEN} \
        scan_layers=true \ # Set to false for unscanned checkpoint
        use_multimodal=false
    ```

  - **Legacy Scripts**: Run the `convert_qwen3_moe.py` script to convert the downloaded Hugging Face weights into the Orbax format required by MaxText.

    ```
    python3 -m maxtext.checkpoint_conversion.standalone_scripts.convert_qwen3_moe\
      --base_model_path /path/to/qwen3_hf_checkpoint\
      --maxtext_model_path gs://your-gcs-bucket/qwen3_maxtext_ckpt\
      --model_size <qwen3-30b-a3b|qwen3-235b-a22b|qwen3-480b-a35b>

    ```

* * * * *

File: tests/end_to_end/tpu/qwen/moe/run_qwen_moe.md (hunk 6 above)
Comment (Rohan-Bierneni) [reply tests/end_to_end/tpu/qwen/moe/run_qwen_moe.md]: I have gone back and fixed the script to initialize properly with default values and allow the user to pass necessary variables. Changes were made in the scripts prefixed with 1_test_qwen3.5_* and the run_qwen_moe.md "correctness validation" section.

File: tests/end_to_end/tpu/qwen/moe/qwen3.5-35b-a3b/2_test_qwen3.5_35b_a3b.sh [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+# 2. Run logit check, pretraining, finetuning, and decoding.
+
+# The golden logit can be generated by:
+# python3 -m tests.assets.logits_generation.generate_hf_golden_logits --model-id=Qwen/Qwen3.5-35B-A3B --output-path=golden_data_qwen3.5-35b-a3b.jsonl --prompts='I love to' --hf-model-path=$local_bf16_path --trust-remote-code=False --hf-load-dtype=bfloat16 
+
+set -ex
+
+export LIBTPU_INIT_ARGS="--xla_tpu_scoped_vmem_limit_kib=65536"
+
+export PYTHONPATH=$PYTHONPATH:$(pwd)/src
```
Comment (shuningjin) [inline tests/end_to_end/tpu/qwen/moe/qwen3.5-35b-a3b/2_test_qwen3.5_35b_a3b.sh]: can remove

File: tests/end_to_end/tpu/qwen/moe/qwen3.5-35b-a3b/2_test_qwen3.5_35b_a3b.sh [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+
+export MODEL_NAME='qwen3.5-35b-a3b'
+export TOKENIZER_PATH='Qwen/Qwen3.5-35B-A3B'
+
+# Installing torch for checkpoint conversion and forward_pass_logit_checker.py
+python3 -m pip install torch --index-url https://download.pytorch.org/whl/cpu
+
+# e.g., $HOME/maxtext/src/maxtext
+export MAXTEXT_PKG_DIR="${MAXTEXT_PKG_DIR:-${MAXTEXT_REPO_ROOT:-$PWD}/src/maxtext}"
```
Comment (shuningjin) [inline tests/end_to_end/tpu/qwen/moe/qwen3.5-35b-a3b/2_test_qwen3.5_35b_a3b.sh]: can remove

File: tests/end_to_end/tpu/qwen/moe/qwen3.5-35b-a3b/2_test_qwen3.5_35b_a3b.sh [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+GOLDEN_LOGITS_DISK_LOCATION="/deps/tests/assets/golden_logits/golden_data_${MODEL_NAME}.jsonl"
+if [ ! -f "${GOLDEN_LOGITS_DISK_LOCATION}" ]; then
+  GOLDEN_LOGITS_PATH="gs://maxtext-test-assets/golden_data_${MODEL_NAME}.jsonl"
+  GOLDEN_LOGITS_DISK_LOCATION=/tmp/golden_data.jsonl
+  gcloud storage cp ${GOLDEN_LOGITS_PATH} ${GOLDEN_LOGITS_DISK_LOCATION}
+fi
+
+python3 -m tests.utils.forward_pass_logit_checker ${MAXTEXT_CONFIGS_DIR:-${MAXTEXT_REPO_ROOT:-$PWD}/configs/base.yml base_output_directory=${BASE_OUTPUT_PATH} run_name=forward_logits_check load_parameters_path=${SCANNED_CKPT_PATH} scan_layers=true attention=dot_product per_device_batch_size=1 model_name=${MODEL_NAME} max_prefill_predict_length=4 max_target_length=4 async_checkpointing=false sparse_matmul=True ici_fsdp_parallelism=1 ici_expert_parallelism=-1 weight_dtype=float32 dtype=float32 activations_in_float32=true matmul_precision=highest float32_logits=true float32_qk_product=true --golden_logits_path=${GOLDEN_LOGITS_DISK_LOCATION} --atol=1.5 --rtol=1.5 --max_kl_div=0.2
```
Comment (shuningjin) [inline tests/end_to_end/tpu/qwen/moe/qwen3.5-35b-a3b/2_test_qwen3.5_35b_a3b.sh]: let's use
```
${MAXTEXT_CONFIGS_DIR:-${MAXTEXT_REPO_ROOT:-$PWD}/src/maxtext/configs}/base.yml
```
to be consistent with other files: https://screenshot.googleplex.com/C6LjNDi89Wv5AKZ

File: tests/end_to_end/tpu/qwen/moe/run_qwen_moe.md
Comment (shuningjin) [inline tests/end_to_end/tpu/qwen/moe/run_qwen_moe.md]: Since qwen3-moe is used very often, let's update to `to_maxtext` example as well. I believe it is well-supported

```
    JAX_PLATFORMS=cpu python3 -m maxtext.checkpoint_conversion.to_maxtext src/maxtext/configs/base.yml \
        model_name=<qwen3-30b-a3b|qwen3-235b-a22b|qwen3-480b-a35b> \
        base_output_directory=gs://your-gcs-bucket/qwen3_maxtext_ckpt \
        hf_access_token=${HF_TOKEN} \
        scan_layers=true \ # Set to false for unscanned checkpoint
        hf_model_path=${HF_MODEL_PATH} # Optional
```

File: tests/end_to_end/tpu/qwen/moe/run_qwen_moe.md [hunk 10]
```diff
@@ -133,4 +156,40 @@ export MAXTEXT_CHECKPOINT_PATH=gs://your-gcs-bucket/qwen3-480b-a35b_maxtext_ckpt
 
 # Execute the validation script
 bash tests/end_to_end/tpu/qwen/moe/qwen3-480b-a35b/1_test_qwen3_480b_a35b.sh
+```
+
+### Qwen3.5-35B-A3B
+
+Bash
+
+```
```
Comment (shuningjin) [inline tests/end_to_end/tpu/qwen/moe/run_qwen_moe.md]: nit: change to?
```
```bash
```
same for other places

**Review by shuningjin — APPROVED** [review]
Thanks for the update!

File: tests/end_to_end/tpu/qwen/moe/qwen3.5-35b-a3b/2_test_qwen3.5_35b_a3b.sh (hunk 7 above)
Comment (Rohan-Bierneni) [reply tests/end_to_end/tpu/qwen/moe/qwen3.5-35b-a3b/2_test_qwen3.5_35b_a3b.sh]: Removed the pythonpath but kept the xla flag arg since required for the workload
